### PR TITLE
gh-630 - hllp aggregator now works with any p and sp values

### PR DESCRIPTION
**Pull request by p013570**

### CODE REVIEW

**Comment by p013570** [comment]
I have written tests for the new functionality, but the test coverage has gone down slightly due to a couple extra lines to catch an IOException that could be thrown by HyperLogLogPlus. It is not simple to invoke this exception so I don't think it is necessary to test it.

**Comment by gaffer01** [comment]
Agreed.

File: library/sketches-library/src/main/java/uk/gov/gchq/gaffer/sketches/function/aggregate/HyperLogLogPlusAggregator.java [hunk 1]
```diff
@@ -42,12 +46,18 @@ public void init() {
...
-                sketch = new HyperLogLogPlus(5, 5);
-            }
-            try {
-                sketch.addAll(input);
-            } catch (CardinalityMergeException exception) {
-                throw new RuntimeException("An Exception occurred when trying to aggregate the HyperLogLogPlus objects", exception);
+                try {
+                    sketch = HyperLogLogPlus.Builder.build(input.getBytes());
+                } catch (IOException e) {
+                    LOGGER.warn("Unable to clone hyper log log plus object", e);
```
Comment (gaffer01) [inline library/sketches-library/src/main/java/uk/gov/gchq/gaffer/sketches/function/aggregate/HyperLogLogPlusAggregator.java]: Can this be written as "Unable to clone a HyperLogLogPlus object".

File: library/sketches-library/src/main/java/uk/gov/gchq/gaffer/sketches/function/aggregate/HyperLogLogPlusAggregator.java [hunk 2]
```diff
@@ -42,12 +46,18 @@ public void init() {
...
             if (null == sketch) {
-                sketch = new HyperLogLogPlus(5, 5);
-            }
-            try {
-                sketch.addAll(input);
-            } catch (CardinalityMergeException exception) {
-                throw new RuntimeException("An Exception occurred when trying to aggregate the HyperLogLogPlus objects", exception);
+                try {
+                    sketch = HyperLogLogPlus.Builder.build(input.getBytes());
+                } catch (IOException e) {
```
Comment (gaffer01) [inline library/sketches-library/src/main/java/uk/gov/gchq/gaffer/sketches/function/aggregate/HyperLogLogPlusAggregator.java]: Should be "final IOException e" to be consistent with other parts of the code.

File: library/sketches-library/src/main/java/uk/gov/gchq/gaffer/sketches/function/aggregate/HyperLogLogPlusAggregator.java [hunk 3]
```diff
@@ -42,12 +46,18 @@ public void init() {
...
+                    LOGGER.warn("Unable to clone hyper log log plus object", e);
+                    sketch = input;
+                }
+            } else {
+                try {
+                    sketch.addAll(input);
+                } catch (CardinalityMergeException exception) {
```
Comment (gaffer01) [inline library/sketches-library/src/main/java/uk/gov/gchq/gaffer/sketches/function/aggregate/HyperLogLogPlusAggregator.java]: Should be "final CardinalityMergeException e" to be consistent with other parts of the code.

**Comment by p013570** [comment]
@gaffer01 - I've made the suggested changes.

**Pull request merged**